api/tests/unit/mappers/test_unit_mappers_dynamodb.py

```diff
@@ -0,0 +1,203 @@
+) -> None:
+    # Given
+    document = {"field": "foobar"}
+    map_identity_to_engine_mock = mocker.patch(
+        "mappers.dynamodb.map_identity_to_engine"
+    )
+    map_engine_identity_to_identity_document_mock = mocker.patch(
+        "mappers.dynamodb.map_engine_identity_to_identity_document"
+    )
+    map_identity_to_engine_mock.return_value = mock_engine_model
+    map_engine_identity_to_identity_document_mock.return_value = document
+
+    # When
+    result = dynamodb.map_identity_to_identity_document(identity)
+
+    # Then
+    assert result == document
+    map_identity_to_engine_mock.assert_called_once_with(identity)
+    map_engine_identity_to_identity_document_mock.assert_called_once_with(
+        mock_engine_model,
+    )
+
+
+def test_map_value_to_document_value__return_expected() -> None:
+    # Given
+    now = datetime.now()
+    uuid_value = uuid.uuid4()
+
+    class Unknown:
+        def __str__(self) -> str:
+            return "bing"
+
+    class TestModel(BaseModel):
+        value_list: Optional[List[Any]] = None
+        value: str = Field(default_factory=lambda: "pydantic works")
+
+    value_list = [
+        TestModel(value_list=[12.2, False, {"key": [TestModel()], "uuid": uuid_value}]),
+        None,
+        True,
+        now,
+        "as is",
+        42,
+        Unknown(),
+    ]
+
+    mv_fs_value = MultivariateFeatureStateValueModel(
+        percentage_allocation=50,
+        multivariate_feature_option=MultivariateFeatureOptionModel(
+            value="foobar",
+        ),
+    )
+    custom_engine_container = MultivariateFeatureStateValueList([mv_fs_value])
+
+    model = TestModel(value_list=value_list, value="outer_value")
+    value = {
+        "outer_key": model,
+        "uuid": uuid_value,
+        "custom_list": custom_engine_container,
+    }
+
+    expected_value = {
+        "outer_key": {
+            "value": "outer_value",
+            "value_list": [
+                {
+                    "value": "pydantic works",
+                    "value_list": [
+                        Decimal("12.2"),
+                        False,
+                        {
+                            "key": [{"value": "pydantic works", "value_list": None}],
+                            "uuid": str(uuid_value),
+                        },
+                    ],
+                },
+                None,
+                True,
+                now.isoformat(),
+                "as is",
+                Decimal("42"),
+                "bing",
+            ],
+        },
+        "uuid": str(uuid_value),
+        "custom_list": [
+            {
+                "id": None,
+                "multivariate_feature_option": {"id": None, "value": "foobar"},
+                "mv_fs_value_uuid": str(mv_fs_value.mv_fs_value_uuid),
+                "percentage_allocation": Decimal("50.0"),
+            }
+        ],
+    }
+
+    # When
+    result = dynamodb._map_value_to_document_value(value)
```

Comment:
Is there any way that we can avoid testing a 'private' method directly?

Reply:
Done.